### PR TITLE
implement MulitModalLinkChooser

### PR DESCRIPTION
**Pull request by simei94**
Extract functionality of FacilitesUtils into MultiModalLinkChooserDefaultImpl. This enables individual implementations of the functionality, which may be necessary in several scenarios

### CODE REVIEW

File: matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            // this is the case where the postal address link is NOT in the subnetwork, i.e. does NOT serve the desired mode,
+            // OR the facility does not have a street address link in the first place.
+
+            if( facility.getCoord()==null ) {
+                throw new RuntimeException("link for facility cannot be determined when neither facility link id nor facility coordinate given") ;
+            }
+
+            accessActLink = NetworkUtils.getNearestLink(network, facility.getCoord()) ;
+            if ( accessActLink == null ) {
+                int ii = 0 ;
```
Comment (Janekdererste) [inline matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java:44]: What is this for loop doing? It looks like it is taking 10 links from the network and printing an error message for them?

File: matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import org.apache.log4j.Logger;
+import org.matsim.api.core.v01.Id;
+import org.matsim.api.core.v01.network.Link;
+import org.matsim.api.core.v01.network.Network;
+import org.matsim.core.gbl.Gbl;
+import org.matsim.core.network.NetworkUtils;
+import org.matsim.facilities.FacilitiesUtils;
+import org.matsim.facilities.Facility;
+
+class MultimodalLinkChooserDefaultImpl implements MultimodalLinkChooser {
```
Comment (Janekdererste) [inline matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java:12]: So, this is some complicated logic. I don't see any unit test for this. Are you testing this implicitly somewhere else?

Comment (simei94) [reply matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java:12]: Basically it is tested in the Leipzig [scenario](https://github.com/matsim-scenarios/matsim-leipzig/commit/4cb6fbcdb0fca84185747b1f1220bb68f224b435).

File: matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java (hunk 1 above)
Comment (vsp-gleich) [reply matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java:44]: This is code copied over here from FacilitiesUtils (where we set the original code to deprecated). I have to admit that we just copied and now I'm wondering too whether this is just some debug output committed by accident into FacilitiesUtils.

File: matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java (hunk 2 above)
Comment (vsp-gleich) [reply matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java:12]: I think we need some test in matsim-libs, too

File: matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java (hunk 1 above)
Comment (vsp-gleich) [reply matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java:44]: I talked to @kainagel about these lines of code. They have a debugging purpose. I added a comment in #2051 to explain.

File: matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java (hunk 2 above)
Comment (Janekdererste) [reply matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java:12]: > Basically it is tested in the Leipzig [scenario](https://github.com/matsim-scenarios/matsim-leipzig/commit/4cb6fbcdb0fca84185747b1f1220bb68f224b435).

That's not sufficient I think.

File: matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java (hunk 1 above)
Comment (Janekdererste) [reply matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java:44]: Ah, this makes it clearer. Copy paste development is the best kind of development 🙃

File: matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java (hunk 2 above)
Comment (simei94) [reply matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java:12]: I will write a test for this class, which covers all the possible outcomes of the logic.

Comment (Janekdererste) [reply matsim/src/main/java/org/matsim/core/router/MultimodalLinkChooserDefaultImpl.java:12]: Thanks for doing this. Sorry that you have to do it now. This should have been there in the old place already.

**Pull request merged**